core/src/trezor/wire/thp/interface_context.py

```
@@ -54,186 +45,418 @@ class ThpContext:
 
-        # read and handle the packet using its `InterfaceContext`
-        iface_ctx._iface.read(self._packet_buf, 0)
-        return await iface_ctx.handle_packet(self._packet_view[:packet_len])
+    async def close(self) -> None:
+        try:
+            close_tasks = (
+                loop.spawn(iface_ctx.close()) for iface_ctx in self._iface_ctxs
+            )
+            for task in close_tasks:
+                await task
+        except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)